lighthouse-cli/test/fixtures/perf/delayed-element.js

```diff
@@ -6,6 +6,24 @@
 
 /* eslint-disable */
 
+/** Create long tasks on the main thread. */
+function stall(ms) {
+  const start = performance.now();
+  while (performance.now() - start < ms) ;
+}
+
+/** Render large number of elements to fill up the backend node cache. */
+function rerender(iterations) {
```

Comment:
might this be easier to read as an async function with a for loop, instead of a recursive promise function? the complexity then would just contained to `requestAnimationFrame` being wrapped in a Promise.